pygmt/base_plotting.py

```diff
@@ -318,24 +318,59 @@ def grdcontour(self, grid, **kwargs):
 
     @fmt_docstring
     @use_alias(
-        R="region",
-        J="projection",
-        W="pen",
+        A="img_out",
         B="frame",
-        I="shading",
         C="cmap",
+        D="img_in",
+        E="dpi",
+        G="bit_color",
+        I="shading",
+        J="projection",
+        M="monochrome",
+        N="no_clip",
+        Q="nan_transparent",
+        R="region",
+        U="timestamp",
+        V="verbose",
         X="xshift",
         Y="yshift",
+        n="interpolation",
         p="perspective",
         t="transparency",
         x="cores",
     )
     @kwargs_to_strings(R="sequence", p="sequence")
     def grdimage(self, grid, **kwargs):
         """
-        Project grids or images and plot them on maps.
-
-        Takes a grid file name or an xarray.DataArray object as input.
+        Project and plot grids or images.
+
+        Reads a 2-D grid file and produces a gray-shaded (or colored) map by
+        building a rectangular image and assigning pixels a gray-shade (or
+        color) based on the z-value and the CPT file. Optionally, illumination
+        may be added by providing a file with intensities in the (-1,+1) range
+        or instructions to derive intensities from the input data grid. Values
+        outside this range will be clipped. Such intensity files can be created
+        from the grid using `grdgradient` and, optionally, modified by
+        `grdmath` or `grdhisteq`. A third alternative is available when GMT is
+        build with GDAL support. Pass **image** which can be an image file
+        (geo-referenced or not). In this case the image can optionally be
+        illuminated with the file provided via the *shading* option. Here, if
+        image has no coordinates then those of the intensity file will be used.
```

Comment:
What do you think about taking out this sentence about *image*, related to the change at https://github.com/GenericMappingTools/pygmt/pull/620/files#r492612750.

Reply:
> What do you think about taking out this sentence about _image_, related to the change at https://github.com/GenericMappingTools/pygmt/pull/620/files#r492612750.

The changes are big. It's unclear to me what you mean by "taking out this sentence about _image_"?

Reply:
Oops, wrong link. Should be https://github.com/GenericMappingTools/pygmt/pull/620#discussion_r492612750. The part where we took out `grid | image`.

Reply:
The reason to take out `grid | image` is, it's a little unclear what `grid | image`, and for the `grid` argument, we already mention that `grid` could be a image:

>            The file name or a DataArray containing the input 2-D gridded data
>            set or image to be plotted

Reply:
Yes, agree that taking `grid | image` is ok. But would this lead to confusion about this sentence:

```
        Pass **image** which can be an image file
        (geo-referenced or not). In this case the image can optionally be
        illuminated with the file provided via the *shading* option. Here, if
        image has no coordinates then those of the intensity file will be used.
```

I suppose we could just keep it to keep it in line with upstream GMT?

Reply:
That's why I said these sentences don't read well in https://github.com/GenericMappingTools/pygmt/pull/620#discussion_r492611273.

Perhaps we can change:
> A third alternative is available when GMT is build with GDAL support. Pass image which can be an image file (geo-referenced or not). In this case the image can optionally be illuminated with the file provided via the -I option. Here, if image has no coordinates then those of the intensity file will be used.

to

> If GMT is built with GDAL support, *grid* can be an image file (geo-referenced or not). In this case the image can optionally be illuminated with the file provided via the -I option. Here, if image has no coordinates then those of the intensity file will be used.

Reply:
Looks good enough (mentioning that *image* can be used with GDAL support). I'll commit the change.